diffsims/generators/rotation_list_generators.py

```diff
@@ -159,83 +161,521 @@ def get_grid_around_beam_direction(beam_rotation, resolution, angular_range=(0,
+        points_in_cartesians = get_uv_sphere_mesh_vertices(resolution)
+    elif mesh == "normalized_cube":
+        # special case: hexagon is a very small slice and 001 point can
+        # be isolated. Hence we increase resolution to ensure minimum angle.
+        if crystal_system == "hexagonal":
+            resolution = np.rad2deg(
+                    np.arctan(np.tan(np.deg2rad(resolution))/np.sqrt(2)))
+        points_in_cartesians = get_cube_mesh_vertices(resolution,
+                                                      grid_type="normalized")
+
+    elif mesh == "spherified_cube_edge":
+        # special case: hexagon is a very small slice and 001 point can
+        # be isolated.  Hence we increase resolution to ensure minimum angle.
+        if crystal_system == "hexagonal":
+            resolution = np.rad2deg(
+                    np.arctan(np.tan(np.deg2rad(resolution))/np.sqrt(2)))
+        points_in_cartesians = get_cube_mesh_vertices(resolution,
+                                                      grid_type="spherified_edge")
+    elif mesh == "spherified_cube_corner":
+        points_in_cartesians = get_cube_mesh_vertices(resolution,
+                                                      grid_type="spherified_corner")
+    elif mesh == "icosahedral":
+        points_in_cartesians = get_icosahedral_mesh_vertices(resolution)
+    else:
+        raise NotImplementedError(f"The mesh {mesh} is not recognized. "
+                                  f"Please use: uv_sphere, normalized_cube, "
+                                  f"spherified_cube_edge, "
+                                  f"spherified_cube_corner, icosahedral")
 
     if crystal_system == "triclinic":
         return points_in_cartesians
-
+    # crop to stereographic triangle which depends on crystal system
     corners = crystal_system_dictionary[crystal_system]
     a, b, c = corners[0], corners[1], corners[2]
     if len(a) == 4:
         a, b, c = uvtw_to_uvw(a), uvtw_to_uvw(b), uvtw_to_uvw(c)
 
-    # eliminates those points that lie outside of the streographic triangle
+    # eliminates those points that lie outside of the stereographic triangle
+    sigfig = 7
     points_in_cartesians = points_in_cartesians[
-        np.dot(np.cross(a, b), c) * np.dot(np.cross(a, b), points_in_cartesians.T) >= 0
+        np.round(np.dot(np.cross(a, b), c) * np.dot(np.cross(a, b),
+                 points_in_cartesians.T), sigfig) >= 0
     ]
     points_in_cartesians = points_in_cartesians[
-        np.dot(np.cross(b, c), a) * np.dot(np.cross(b, c), points_in_cartesians.T) >= 0
+        np.round(np.dot(np.cross(b, c), a) * np.dot(np.cross(b, c),
+                 points_in_cartesians.T), sigfig) >= 0
     ]
     points_in_cartesians = points_in_cartesians[
-        np.dot(np.cross(c, a), b) * np.dot(np.cross(c, a), points_in_cartesians.T) >= 0
+        np.round(np.dot(np.cross(c, a), b) * np.dot(np.cross(c, a),
+                 points_in_cartesians.T), sigfig) >= 0
     ]
```

Comment:
Would prefer using `EPSILON = -1e-14` and compare to that, see https://github.com/pyxem/orix/pull/60/files for example